x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -163,6 +167,28 @@ public LocalExecutionPlan plan(PhysicalPlan node) {
             AggregateExec.class,
             a -> a.getMode() == AggregateExec.Mode.FINAL ? new ProjectExec(a.source(), a, Expressions.asAttributes(a.aggregates())) : a
         );
+        if (configuration.pragmas().timeSeriesMode()) {
+            var tsidAttribute = new FieldAttribute(Source.EMPTY, "_tsid", new EsField("_tsid", DataTypes.KEYWORD, Map.of(), true));
+            var timestampAttribute = new FieldAttribute(
+                Source.EMPTY,
+                "@timestamp",
+                new EsField("@timestamp", DataTypes.DATETIME, Map.of(), true)
+            );
+            node = node.transformUp(EsQueryExec.class, source -> {
+                var attrs = new ArrayList<>(source.output());
+                attrs.add(tsidAttribute);
+                attrs.add(timestampAttribute);
+                return new EsQueryExec(
+                    source.source(),
+                    source.index(),
+                    attrs,
+                    source.query(),
+                    source.limit(),
+                    source.sorts(),
+                    source.estimatedRowSize()
+                );
+            });
```

Comment:
I'd love someone from the QL half of the team to look at this bit.

Reply:
The usage of tsid of timestamp fields is internal. I will update the PR to remove the transformation and create a subclass of `EsQueryExec` for when the query pragma is enabled.

> we can follow-up with projecting away the attributes in a different PR (similar to what happens with EsQueryExec#DOC_ID_FIELD

👍